pkg/restic/restorer.go

```diff
@@ -85,7 +84,7 @@ func (r *restorer) RestorePodVolumes(restore *arkv1api.Restore, pod *corev1api.P
 		return nil
 	}
 
-	repo, err := getReadyRepo(r.repoLister, restore.Namespace, pod.Namespace)
+	repo, err := r.repoEnsurer.EnsureRepo(r.ctx, restore.Namespace, pod.Namespace)
```

Comment:
Would we ever end up creating a repo here? I'm thinking that even if we do, the data and buckets are in an invalid state already.

Reply:
This has the potential to create the restic repo. We shouldn't be doing that as part of a restore.

Reply:
How else would you handle the case where a user installs Ark in a fresh cluster and immediately wants to restore an existing backup with restic snapshots?

Reply:
My bad, I wasn't thinking about the restore half of the equation (that's what happens when I try to wrap up a code review at the end of the day after a long meeting!).